examples/asset/remote_asset_loader.rs

```diff
@@ -0,0 +1,167 @@
+//! Implements a remote asset loader.
+//! An [`AssetReader`] is what the asset server uses to read the raw bytes of assets.
+//! This example also showcase how to use [`AssetWriter`], thus will won't work on
+//! wasm or Android, since those platforms doesn't have [`AssetWriter`].
+
+use bevy::{
+    asset::io::{
+        AssetReader, AssetReaderError, AssetSource, AssetSourceId, AssetWriter, PathStream, Reader,
+    },
+    prelude::*,
+    utils::BoxedFuture,
+};
+use bevy_internal::asset::io::AssetWriterError;
+use futures_lite::AsyncRead;
+use std::{path::Path, sync::Arc};
+
+/// A remote asset loader implementation that will try to read asset local and, if not found,
+/// download it remotely from a CDN.
+struct RemoteAssetLoader {
+    cdn_prefix: String,
+    reader: Box<dyn AssetReader>,
+    writer: Box<dyn AssetWriter>,
+}
+
+impl RemoteAssetLoader {
+    fn new() -> Self {
+        // Create the root directory if it doesn't exists on local disk.
+        let create_root_dir = true;
+        let writer = AssetSource::get_default_writer("assets".to_string())(create_root_dir)
+            .expect("Current platform doesn't support asset writing.");
+        let reader = AssetSource::get_default_reader("assets".to_string())();
+
+        // This is the CDN which will be downloaded the font when missing localy.
+        let cdn_prefix =
+            "https://github.com/google/fonts/raw/main/ofl/protestrevolution/".to_string();
```

Comment:
Perhaps we should use the [Bevy GitHub](https://github.com/bevyengine/bevy/raw/main/assets/fonts/FiraMono-Medium.ttf) repo instead? Just avoids any possible conversations around attribution, licensing, etc. since anything within the repo is already fully vetted and controlled.

Reply:
But where? This can't be on bevyengine/bevy since it is already downloading from it. Maybe on bevy website repo? Anyway, Google Fonts repo is pretty stable.

Reply:
What I'm suggesting is using the existing `fonts` folder as the CDN, but trying to access it from an alternative local folder name. So:

1. Try to access "saved_assets/fonts/FiraMono-Medium.ttf"
2. If it doesn't exist, fetch from https://github.com/bevyengine/bevy/raw/main/assets and save

I do agree that the link you've included is fine, this is purely to ensure links go to Bevy controlled areas.